account-kit/react/src/components/auth/context.ts

```diff
@@ -56,7 +56,20 @@ export function useAuthContext<
   TType extends AuthStep["type"] | undefined = AuthStep["type"] | undefined
 >(type?: TType): AuthContextType<TType>;
 
-export function useAuthContext(
+/**
+ * A custom hook that provides the authentication context based on the specified authentication step type. It ensures that the hook is used within an `AuthModalProvider` and throws an error if the context is not available or if the current auth step type does not match the expected type.
+ *
+ * @example
+ * ```tsx
```

Comment:
nit: 
```suggestion
 * ```tsx twoslash
```

this will enable us to ensure correctness of the examples when we build the docs because the examples will get typechecked